android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboardPickerAdapter.java

```diff
@@ -46,29 +47,8 @@ public KMKeyboardPickerAdapter(final Context context, Dataset.Keyboards adapter)
       // Yeah, so this is a MASSIVE hack.  Right now, it's either this or refactor up to 60
       // separate references to keyboardList within KeyboardPickerActivity.  Yikes.
       public List<Keyboard> selectFrom(Dataset.Keyboards adapter, Void dummy) {
-        List<HashMap<String, String>> kbdMapList = KeyboardPickerActivity.getKeyboardsList(context);
-        List<Keyboard> kbdList = new ArrayList<>(kbdMapList.size());
-
-        for(HashMap<String, String> kbdMap: kbdMapList) {
-          boolean isNewKeyboard = kbdMap.containsKey(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD) &&
-            kbdMap.get(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD).equals(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD);
-
-          Keyboard k = new Keyboard(
-            kbdMap.get(KMManager.KMKey_PackageID),
-            kbdMap.get(KMManager.KMKey_KeyboardID),
-            kbdMap.get(KMManager.KMKey_KeyboardName),
-            kbdMap.get(KMManager.KMKey_LanguageID),
-            kbdMap.get(KMManager.KMKey_LanguageName),
-            MapCompat.getOrDefault(kbdMap, KMManager.KMKey_Version, "1.0"),
-            MapCompat.getOrDefault(kbdMap, KMManager.KMKey_CustomHelpLink, ""),
-            isNewKeyboard,
-            MapCompat.getOrDefault(kbdMap, KMManager.KMKey_Font, null),
-            MapCompat.getOrDefault(kbdMap, KMManager.KMKey_OskFont, null)
-          );
-          kbdList.add(k);
-        }
-
-        return kbdList;
+        // Return the keyboards list
+        return adapter.asList();
```

Comment:
Assuming this is functioning properly, we can finally drop that "MASSIVE HACK" comment/disclaimer.